.github/workflows/gatk-tests.yml

```diff
@@ -0,0 +1,311 @@
+        uses: google-github-actions/setup-gcloud@v0
+
+      - name: build test jars
+        run: ./gradlew clean shadowTestClassJar shadowTestJar
+
+      - name: set up test folders and links
+        run: |
+          mkdir ./testJars;
+          echo $( find ./build/libs/ -name "gatk*test.jar");
+          cp -r $( find ./build/libs/ -name "gatk*test.jar" ) ./testJars;
+          cp -r $( find ./build/libs/ -name "gatk*testDependencies.jar" ) ./testJars;
+          mkdir -p build/reports/;
+          chmod -R a+w build/reports/;
+          cp scripts/docker/dockertest.gradle .
+
+      - name: pull docker
+        run: docker pull ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}:${{ github.sha }}
+
+      - name: pull lfs files
+        run: git lfs pull
+
+      - name: run-docker-tests
+        id: docker-jacocco-tests
+        run: |
+          docker run \
+            -v $(pwd):/gatkCloneMountPoint:cached \
+            -v $(pwd)/testJars:/jars:cached \
+            --rm \
+            -e "scala.version=${SCALA_VERSION}" \
+            -e "TEST_VERBOSITY=minimal" \
+            -e "TEST_TYPE=${TEST_TYPE}" \
+            -t  ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}:${{ github.sha }} \
+            bash --init-file /gatk/gatkenv.rc /root/run_unit_tests.sh;
+          TEST_EXIT_VALUE=$?;
+          $( exit ${TEST_EXIT_VALUE} );
+          sudo chmod -R a+w build/reports/;
+          mkdir build/reports/tests/test \
+           && cp -rp build/reports/tests/testOnPackagedReleaseJar/* build/reports/tests/test \
+           && rm -r build/reports/tests/testOnPackagedReleaseJar;
+
+      - uses: /broadinstitute/gatk/.github/actions/upload-results@lb_github_actions
+        if: always()
+        with:
+          warnPR: ${{ github.event_name == 'pull_request' && steps.docker-jacocco-tests.outcome != 'success' }}
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
+          jdk-version: ${{ matrix.Java }}
+          test-type: docker ${{ matrix.testType }}
+          job-matrix-id: ${{ github.run_id }}.${{ strategy.job-index }}
+          repo-path: ${{ github.ref_name }}_${{ github.run_id }}.${{ strategy.job-index }}
+          bot-comment-key: ${{ secrets.GATK_BOT_COMMENT_KEY }}
+          identifier: Java ${{ matrix.Java }} test on docker ${{matrix.testType}}
+
+
+  wdlGen:
+    runs-on: ubuntu-latest
+    name: Test WDL Generation
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch: 0
+      - name: Set up java 8
+        uses: actions/setup-java@v2
+        with:
+          java-version: '8'
+          distribution: 'adopt'
+          cache: gradle
+      - uses: /broadinstitute/gatk/.github/actions/install-cromwell@lb_github_actions
+        with:
+          CROMWELL_VERSION: ${{ env.CROMWELL_VERSION }}
+
+      - name: Grant execute permission for gradlew
+        run: chmod +x gradlew
+
+      - name: Run WDL Generation Test
+        run: ./gradlew gatkWDLGenValidation
+
+  wdlTests:
```

Comment:
This whole section is gross, which his not your fault since it was even more gross before.  We should maybe fix it sometime.